### ISSUE TITLE
Document the separate MPS proposal

### ISSUE THREAD

**Issue by kwxm** [input-output-hk/plutus#1827]
In #1801, @michaelpj implemented a proposal for separate monetary policy scripts (see the "A possible simplification to the multicurrency system" email thread).

Just so we don't forget, this should be reflected in 

  * The EUTXO document
  * The multicurrency paper
  * The Agda formalisation of EUTXO-2

**Comment by kwxm** [comment]
I shall do this based on the implementation and then @michaelpj can take a look to see if what I come up with is sensible: having two of us look at it may help to turn up any issues.  

However, there's also a proposal for spending policies in the pipeline: see #1791. Maybe we should wait for that and then do everything at once.

**Comment by michaelpj** [comment]
Polina and I are having some discussions over the exact format as well, so I guess we shouldn't jump the gun on this.

**Comment by kwxm** [comment]
I think I'll start doing this soon so we can put it in the multicurrency paper.  I'll take a look at the implementation and try to distil what it's doing.  Is anything likely to change in the near future?

**Comment by michaelpj** [comment]
Let me summarise the current state. There are some divergences.

----

First, about the location of the scripts.

In the implementation, we follow my original proposal:
- There is an additional field (called `forgeScripts`, but the name doesn't matter much).
- All the forge scripts are run unconditionally and must pass.
- When checking the forge field, we check that a script corresponding to every currency ID appears in `forgeScripts`.

In Polina's spec, we've diverged a bit:
- To save space, we don't list the `forgeScripts` separately, but rather assume that we can iterate over the `forge` field to find all the ones that we need.
- Scripts are run only if they appear as currency IDs in the `forge` field (my version would allow you to add "redundant" ones).

I discussed this with Manuel, and we think that for simplicity we should present the original version in the paper. The difference is only really representational, so we might as well pick the one that's more explicit for now.

-----

Second, about redeemers.

Currently the implementation does not support redeemers for MPSs, and Polina's spec does. 

I'm not sure if we should include them in the paper: we don't have any usecases for them, so they'd be a bit of dangling feature. They're easy enough to add in later if we want them, so I think we should just leave them out for now.
